docs(review): Reviewed docs K8s X LB (#2327)
scaleway/docs-content: containers/kubernetes/reference-content/using-load-balancer-annotations.mdx

---
meta:
  title: Using Scaleway Load Balancer annotations
  description: This page explains how to use Scaleway Load Balancer annotations
content:
  h1: Using Scaleway Load Balancer annotations
  paragraph: This page explains how to use Scaleway Load Balancer annotations
tags: kubernetes load-balancer load-balancer-annotations annotations
dates:
  validation: 2023-10-23
  posted: 2021-08-12
categories:
  - kubernetes
---

## Overview

In Kubernetes, annotations are a way to attach metadata to objects, like pods, services, and more. These annotations are key-value pairs that can be useful for tools and libraries interacting with these objects. These annotations are used to influence the behavior, settings, or configurations of the provisioned load balancer resource.

> When you [create a Load Balancer](/containers/kubernetes/reference-content/kubernetes-load-balancer/) for your Kubernetes cluster, it will be created with a default configuration, unless you define its configuration parameters via **annotations**. Load Balancer annotations let you configure parameters such as the balancing method health check settings and more.

Reviewer: @RoRoJ it's weird but in the staging preview I see an extra ] after balancing method.
![Capture d’écran 2023-11-06 à 17 43 47](https://github.com/scaleway/docs-content/assets/109532009/a2ec7d89-6330-4475-a08f-5481389ad753)

Reviewer 2: The staging is not dynamically updated with the base branch - we've corrected this since I deployed the staging.

    spec:
      type: LoadBalancer
      ports:
      - port: 8000
        name: http
        targetPort: 8000
      selector:
        app: mydeployment
    ```

You can then create your Load Balancer with the `kubectl create -f <name-of-manifest-file>.yaml` command. 

When you examine your Load Balancer's backend configuration and health check information in the Scaleway console, you will see that the balancing method (aka forward port algorithm) and health check interval (aka health check delay) of its backend reflect the values set in the annotations.

## Modifying your Load Balancer's configuration via annotations after creation

To modify your Load Balancer's configuration with annotations after creation, you can use the `kubectl patch` command. This updates the Load Balancer according to the arguments given.

In this example, we create the following file to hold our patch, and call it `annotations-patch.yaml`:

```yaml
metadata:
  annotations:
    service.beta.kubernetes.io/scw-loadbalancer-forward-port-algorithm: "first"
    service.beta.kubernetes.io/scw-loadbalancer-health-check-timeout: "30s"
```

> You can then tell the CCM to put this patch into effect with the following command:

Reviewer 3: ```suggestion
You can then make the CCM execute this patch with the following command:
```

```
kubectl patch svc myloadbalancer --type merge --patch-file annotations-patch.yaml
```

When you examine your Load Balancer's backend configuration and health check information in the Scaleway console, you will see that the balancing method (aka forward port algorithm) and health check timeout of its backend reflect the values set in the patched annotations.

> ```yaml

Reviewer 2: Thanks, fixed, another copy/paste issue.
